byway/src/main/webapp/track-interests.js

```diff
@@ -12,13 +12,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+let interestsChosen = new Set();
+let tripQuery;
+
 if(document.readyState === 'loading') {
-  document.addEventListener('DOMContentLoaded', loadButtons);
+  document.addEventListener('DOMContentLoaded', loadContent);
 } else {
+  loadContent();
+}
+
+/**
+ * Loads the content of the page by loading buttons
+ * and the form tracking buttons selected.
+ */
+function loadContent() {
+  loadId();
   loadButtons();
+  loadForm();
 }
 
-let interestsChosen = new Set();
+/* Parse url to retrieve the trip id. */
+function loadId() {
+  let queryString = location.search.substring(1);
+  let separatedData = queryString.split("&");
+  tripQuery = separatedData[0];
```

Comment:
There are functions available for parsing query parameters, namely `URLSearchParams`. Don't try to split up data like that yourself.

Remember that URLs are user-visible, and anyone can put whatever they want in them. What happens if you try to split like this and there are query parameters that you didn't expect? What if you add more parameters in the future? Remember order in query parameters generally doesn't matter, so grabbing by index is a bad idea.

Reply:
Good point, fixed